arangod/RestServer/FlushFeature.cpp

```diff
@@ -112,102 +105,29 @@ arangodb::Result FlushFeature::releaseUnusedTicks(size_t& count,
     TRI_TerminateDebugging("crashing before syncing min tick");
   }
 
-  // WAL tick has to be synced prior to releasing it, if the storage
-  // engine supports it
-  //   engine->waitForSyncTick(minTick);
-
-  TRI_IF_FAILURE("FlushCrashAfterSyncingMinTick") {
-    TRI_TerminateDebugging("crashing after syncing min tick");
-  }
-
   engine.releaseTick(minTick);
 
   TRI_IF_FAILURE("FlushCrashAfterReleasingMinTick") {
     TRI_TerminateDebugging("crashing after releasing min tick");
   }
 
-  return {};
-}
-
-void FlushFeature::validateOptions(
-    std::shared_ptr<options::ProgramOptions> /*options*/) {
-  if (_flushInterval < 1000) {
-    // do not go below 1000 microseconds
-    _flushInterval = 1000;
-  }
-}
-
-void FlushFeature::prepare() {
-  // At least for now we need FlushThread for ArangoSearch views
-  // on a DB/Single server only, so we avoid starting FlushThread on
-  // a coordinator and on agency nodes.
-  setEnabled(!arangodb::ServerState::instance()->isCoordinator() &&
-             !arangodb::ServerState::instance()->isAgent());
-
-  if (!isEnabled()) {
-    return;
-  }
-}
-
-void FlushFeature::start() {
-  {
-    WRITE_LOCKER(lock, _threadLock);
-    _flushThread.reset(new FlushThread(*this, _flushInterval));
-  }
-  DatabaseFeature& dbFeature = server().getFeature<DatabaseFeature>();
-  dbFeature.registerPostRecoveryCallback([this]() -> Result {
-    READ_LOCKER(lock, _threadLock);
-    if (!this->_flushThread->start()) {
-      LOG_TOPIC("bdc3c", FATAL, Logger::FLUSH) << "unable to start FlushThread";
-      FATAL_ERROR_ABORT();
-    } else {
-      LOG_TOPIC("ed9cd", DEBUG, Logger::FLUSH) << "started FlushThread";
-    }
-
-    this->_isRunning.store(true);
+  LOG_TOPIC("2b2e2", DEBUG, arangodb::Logger::FLUSH)
+      << "Flush tick released: '" << minTick << "'"
+      << ", stale flush subscription(s) released: " << stale
+      << ", active flush subscription(s) released: " << active
+      << ", initial engine tick: " << initialTick;
 
-    return {TRI_ERROR_NO_ERROR};
-  });
-}
-
-void FlushFeature::beginShutdown() {
-  // pass on the shutdown signal
-  READ_LOCKER(lock, _threadLock);
-  if (_flushThread != nullptr) {
-    _flushThread->beginShutdown();
-  }
+  return std::make_tuple(active, stale, minTick);
 }
 
 void FlushFeature::stop() {
```

Comment:
Suggest to make it noexcept. In practice if mutex::lock shouldn't throw exception
For example absl::Mutex::lock implemented without exception

So if exception throw-ed it always something really bad like wrong implementation of mutex or race between ctor of mutex and call lock